saleor/graphql/discount/types/promotions.py

```diff
@@ -99,3 +115,65 @@ def resolve_promotion(root: models.PromotionRule, info: ResolveInfo):
     @staticmethod
     def resolve_channels(root: models.PromotionRule, info: ResolveInfo):
         return ChannelsByPromotionRuleIdLoader(info.context).load(root.id)
+
+
+class PromotionEvent(ModelObjectType[models.PromotionEvent]):
+    id = graphene.GlobalID()
+    date = graphene.DateTime(description="Date when event happened.")
+    type = PromotionEventsEnum(description="Promotion event type.")
+    user = graphene.Field(User, description="User who performed the action.")
+    app = graphene.Field(App, description="App that performed the action.")
```

Comment:
Have you considered using union for `app` and `user`? 
Something as we have in `saleor.graphql.payment.types.TransactionEvent` - `created_by` field